packages/lit-dev-content/site/docs/components/lifecycle.md

```diff
@@ -357,22 +357,57 @@ window.onunhandledrejection = function(e) {
 
-#### performUpdate()  {#performupdate}
+This section covers some less-common methods for customizing the update cycle.
 
-Implements the reactive update cycle, calling the other methods, like `shouldUpdate()`, `update()`, and `updated()`.
+#### scheduleUpdate() {#scheduleupdate}
 
-Call `performUpdate()` to immediately process a pending update. This should generally not be needed, but it can be done in rare cases when you need to update synchronously.
+Override `scheduleUpdate()` to customize the timing of the update. `scheduleUpdate()` is called when an update is about to be performed, and by default it calls `performUpdate()` immediately. Override it to defer the update. 
 
-Implement `performUpdate()` to customize the timing of the update cycle. This can be useful for implementing custom scheduling. Note, if `performUpdate()` returns a Promise, the `updateComplete` Promise will await it.
+For example, the following code schedules the update to occur after the next frame paints, which can reduce jank if the update is expensive:
+
+{% switchable-sample %}
+
+```ts
+protected override async scheduleUpdate(): Promise<void> {
+  await new Promise((resolve) => setTimeout(resolve));
+  this.performUpdate();
+}
+```
 
 ```js
-async performUpdate() {
-   await new Promise((resolve) => setTimeout(() => resolve()));
-   return super.performUpdate();
+async scheduleUpdate() {
+  await new Promise((resolve) => setTimeout(resolve));
+  this.performUpdate();
 }
 ```
 
-In this example, the update is performed after paint. This technique can be used to unblock the main rendering/event thread. See the Chrome Dev Summit talk by Justin Fagnani [The Virtue of Laziness](https://www.youtube.com/watch?v=ypPRdtjGooc) for an extended discussion.
+{% endswitchable-sample %}
+
+If you override `scheduleUpdate()`, it's your responsibility to call `performUpdate()`.
+
+{% aside "info" %}
+
+Async function optional.
+
+This example shows an [async function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) which _implicitly_ returns a promise. You can also write `scheduleUpdate()` as a function that _explictly_ returns a `Promise`. In either case, the **next** update doesn't start until the promise returned by `scheduleUpdate()` resolves. 
+
+{% endaside %}
+
+This technique can be used to unblock the main rendering/event thread. See the Chrome Dev Summit talk by Justin Fagnani [The Virtue of Laziness](https://www.youtube.com/watch?v=ypPRdtjGooc) for an extended discussion. This talk shows an early version of Lit, but describes many useful concepts.
+
+#### performUpdate()  {#performupdate}
+
+Implements the reactive update cycle, calling the other methods, like `shouldUpdate()`, `update()`, and `updated()`.
+
+Call `performUpdate()` to immediately process a pending update. This should generally not be needed, but it can be done in rare cases when you need to update synchronously.
```

Comment:
There's a trick here where you need to call `this.isUpdatePending = true` first to get `performUpdate()` to do anything. cc @sorvell

Reply:
Hmm. That did not seem to be the case when I tested the examples, but I may have missed something.

Reply:
Hmm... Interesting point. It says "pending update" ... If someone's enqueued an update, but then deferred it in `scheduleUpdate()`, then `isUpdatePending` will be true and calling `performUpdate()` processes the current pending update (which should set `isPendingUpdate` to false, and cause the eventual, scheduled update to be a no-op. Is that a realistic scenario?

What are the other cases where you could have a pending update but _not_ have `isPendingUpdate()` be true?

Reply:
Also, looking at the current docs for `isUpdatePending`, it says the value should only be read. So if we're going to document this trick here, we should probably also update the API docs.

https://lit.dev/docs/api/ReactiveElement/#ReactiveElement.isUpdatePending

Reply:
I think the current words are correct: "tim immediately process a _pending update_. Could maybe just add "If no update is pending, this method does nothing." And if users _want_ it to do something, just call `requestUpdate()` before `performUpdate()` (instead of any suggestion they should set the flag).

Reply:
When I tried calling `performUpdate()` directly myself it didn't do anything since there was no update pending and the first thing `performUpdate()` does is check `isUpdatePending`: https://github.com/lit/lit/blob/main/packages/reactive-element/src/reactive-element.ts#L1283

@sorvell was there another intended way to provoke an update? `this. requestUpdate(); this.performUpdate()` should also work.

Reply:
I think it makes sense to keep `requestUpdate()` as the canonical way to trigger a pending update. `this.requestUpdate(); this.performUpdate();` (or `this.reactiveProperty = newValue; this.performUpdate();`) can be used if you really need to force a synchronous update. But how often do you need to do this?

I looked for examples in the MWC web components and Shoelace; I didn't see anyone calling `this.performUpdate()` explicitly. On a wider search in GitHub, I saw some examples. Most of these appear to be unnecessary (for example, components that call `performUpdate()` after every property change). 

One component uses the pattern change property, `performUpdate()`, dispatch event, which might make sense... although arguably you could do the same thing by dispatching the event later, from `updated()`, say.

Reply:
As noted, the current way to ensure that you "force an update" is with 2 calls `this.requestUpdate(); this.performUpdate()`. I believe we've previously considered this a rough spot in the API and that we could add a `forceUpdate` method to make it simpler, but my recollection is that we were holding off on that pending feedback on the need for it.

Reply:
Updated to add mention of `requestUpdate()` to force an update.